File: attribute-compiler/src/main/java/pyxis/uzuki/live/attribute/parser/compiler/model/AttrResourceModel.java
package pyxis.uzuki.live.attribute.parser.compiler.model;

import javax.lang.model.element.VariableElement;

import pyxis.uzuki.live.attribute.parser.annotation.AttrColor;
import pyxis.uzuki.live.attribute.parser.annotation.AttrInt;

/**
 * AttributesParser
 * Class: AttrIntModel
 * Created by Pyxis on 3/4/18.
 * <p>
 * Description:
 */

public class AttrColorModel extends BaseAttrModel {
    private String mSource;
    private int mDefValue;

    public AttrColorModel(VariableElement element) {
        super(element);

        AttrColor annotation = element.getAnnotation(AttrColor.class);

        mSource = annotation.value();
        mDefValue = annotation.def();

        findEnclosingClass();
    }

    public String getSource() {
        return mSource;
    }

    public int getDefValue() {
        return mDefValue;
    }
}

File: attribute-parser/src/main/java/pyxis/uzuki/live/attribute/parser/model/AttrResourceModel.kt
package pyxis.uzuki.live.attribute.parser.model

import pyxis.uzuki.live.attribute.parser.annotation.AttrResource
import javax.lang.model.element.VariableElement

/**
 * AttributesParser
 * Class: AttrIntModel
 * Created by Pyxis on 3/4/18.
 *
 *
 * Description:
 */

class AttrResourceModel(element: VariableElement) : BaseAttrModel<AttrResource>(element, AttrResource::class.java) {
    init {
        source = annotation.value
        defValue = annotation.def
    }
}